recipes/pcapplusplus/all/conanfile.py

```diff
@@ -58,15 +77,36 @@ def _configure_sh_script(self):
         }[str(self.settings.os)]
 
     def _patch_sources(self):
-        if not self.options.get_safe("fPIC"):
-            tools.replace_in_file(os.path.join(self._source_subfolder, "PcapPlusPlus.mk.common"),
+        if not self.options.get_safe("fPIC") and self.settings.os != "Windows":
+            files.replace_in_file(self, os.path.join(self._source_subfolder, "PcapPlusPlus.mk.common"),
                                   "-fPIC", "")
         for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
+            files.patch(self, **patch)
 
     def build(self):
         self._patch_sources()
-        with tools.chdir(self._source_subfolder):
+        if self.settings.os == "Windows":
+            self._build_windows()
+        else:
+            self._build_posix()
+
+    def _build_windows(self):
+        vs_version = self._get_vs_version()
+        with files.chdir(self, self._source_subfolder):
+            config_args = [
+                "configure-windows-visual-studio.bat",
+                "--pcap-sdk", self.deps_cpp_info["npcap"].rootpath,
+                "--pthreads-home", self.deps_cpp_info["pthreads4w"].rootpath,
+                "--vs-version", vs_version,
+            ]
+            self.run(" ".join(config_args), run_environment=True)
+            msbuild = MSBuild(self)
+            targets =  ['Common++', 'Packet++', 'Pcap++']
+            cmd = msbuild.command(f"mk/{vs_version}/PcapPlusPlus.sln", targets)
+            self.run(cmd + " /p:PlatformToolset=" + tools.msvs_toolset(self))
```

Comment:
I'm not a Conan expert - but is this the best way to run the `msbuild` command? 🤔

Reply:
@seladb No more, toolset should be configured from settings, so part of the profile, otherwise, it will try to detect one: https://github.com/conan-io/conan/blob/fa52c51d58383371a9aece3d2b21d29037d93131/conan/tools/microsoft/toolchain.py#L86

Reply:
@uilianries the problem is that `MSBuild` helper is not actually passing `PlatformToolset` from the settings, it only passes `Configuration` and `Platform`

Reply:
@uilianries using `MSBuild` from `conans.client.build.msbuild` solved the issue.